src/python/py_oiio.cpp

```diff
@@ -219,6 +219,47 @@ oiio_attribute_typed(const std::string& name, TypeDesc type,
 
 
 
+py::object
+make_pyobject(const void* data, TypeDesc type, int nvalues,
+              py::object defaultvalue)
+{
+    if (type.basetype == TypeDesc::INT32)
+        return C_to_val_or_tuple((const int*)data, type, nvalues);
+    if (type.basetype == TypeDesc::FLOAT)
+        return C_to_val_or_tuple((const float*)data, type, nvalues);
+    if (type.basetype == TypeDesc::STRING)
+        return C_to_val_or_tuple((const char**)data, type, nvalues);
+    if (type.basetype == TypeDesc::UINT32)
+        return C_to_val_or_tuple((const unsigned int*)data, type, nvalues);
+    if (type.basetype == TypeDesc::INT16)
+        return C_to_val_or_tuple((const short*)data, type, nvalues);
+    if (type.basetype == TypeDesc::UINT16)
+        return C_to_val_or_tuple((const unsigned short*)data, type, nvalues);
+    if (type.basetype == TypeDesc::INT64)
+        return C_to_val_or_tuple((const int64_t*)data, type, nvalues);
+    if (type.basetype == TypeDesc::UINT64)
+        return C_to_val_or_tuple((const uint64_t*)data, type, nvalues);
+    if (type.basetype == TypeDesc::DOUBLE)
+        return C_to_val_or_tuple((const double*)data, type, nvalues);
+    if (type.basetype == TypeDesc::HALF)
+        return C_to_val_or_tuple((const half*)data, type, nvalues);
+    if (type.basetype == TypeDesc::UINT8 && type.arraylen > 0) {
+        // Array of uint8 bytes
+        // Have to make a copy of the data, because make_numpy_array will
+        // take possession of it.
+        uint8_t* ucdata(new uint8_t[type.arraylen * nvalues]);
+        memcpy(ucdata, data, type.arraylen * nvalues);
```

Comment:
Maybe `std::memcpy`? Recent C++ standards are starting to diverge from C, e.g. functions from <cmath> got constexpr specifiers.

Reply:
But these are dynamic memory and so can't possibly use a constexpr version here anyway.

I dunno, is it considered bad style to use the "non-std::" way to call a traditional C function that's also in C++? I've always just called the C function when they are identical.

Reply:
Yes, it is considered bad style. Prior to C++23 these headers were actually deprecated. C++23 clarifies the status of the headers as "For compatibility with the C standard library" and also [says](https://github.com/cplusplus/draft/pull/5025/files#diff-f46371f476893d323cfcb1383e9a70f07ff1c1f0c217e22294d0bf937aae6794R5893) that "Source files that are not intended to also be valid ISO C should not use any of the C headers."

If interesting, this [paper](http://www7.open-std.org/JTC1/SC22/WG21/docs/papers/2021/p2340r1.html) contains rationale.

Reply:
Interesting!

Ok, I will change this. I'm sure we have this kind of thing scattered all over the place. We'll clean up a bit here and there as we go.

Reply:
Updated, thanks for the suggestion and the education.